app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt

```diff
@@ -147,24 +125,61 @@ class InstallerViewModel(
                                 input.selectedPatches
                             )
                         }
-                    } else {
-                        app.toast(app.getString(R.string.install_app_fail, extra))
                     }
+
+                    installerStatus = pmStatus
+                    showInstallerDialog = true
                 }
+            }
+        }
+    }
 
+    private val uninstallBroadcastReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            when (intent?.action) {
                 UninstallService.APP_UNINSTALL_ACTION -> {
+                    val extraStatus =
+                        intent.getIntExtra(UninstallService.EXTRA_UNINSTALL_STATUS, -999)
+                    val extraStatusMessage =
+                        intent.getStringExtra(UninstallService.EXTRA_UNINSTALL_STATUS_MESSAGE)
+
+                    if (extraStatus != PackageInstaller.STATUS_FAILURE_ABORTED) {
+                        app.toast(app.getString(R.string.uninstall_app_fail, extraStatusMessage))
+                    }
                 }
             }
         }
     }
 
     init {
-        ContextCompat.registerReceiver(app, installBroadcastReceiver, IntentFilter().apply {
-            addAction(InstallService.APP_INSTALL_ACTION)
-            addAction(UninstallService.APP_UNINSTALL_ACTION)
-        }, ContextCompat.RECEIVER_NOT_EXPORTED)
+        // TODO: navigate away when system-initiated process death is detected
+        //  because it is not possible to recover from it.
+        viewModelScope.launch {
+            installedApp = installedAppRepository.get(packageName)
+        }
+
+        ContextCompat.registerReceiver(
+            app, installBroadcastReceiver,
+            IntentFilter(InstallService.APP_INSTALL_ACTION),
+            ContextCompat.RECEIVER_NOT_EXPORTED
+        )
+
+        ContextCompat.registerReceiver(
+            app, uninstallBroadcastReceiver,
+            IntentFilter(UninstallService.APP_UNINSTALL_ACTION),
+            ContextCompat.RECEIVER_NOT_EXPORTED
+        )
```

Comment:
Why do the receivers need to be separated?

Reply:
I am not sure of any other implementation.

Reply:
It is possible to handle both with a single receiver. That is what was done before. What I am asking is: Is there a reason to separate them?

Reply:
I am unaware of an implementation that allows handling both with a single receiver. If possible, please link to one as a reference.

Reply:
[This](https://github.com/ReVanced/revanced-manager/blob/ac561e7acaf190605d02ff22ba49df6cd655a8e9/app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt#L130-L166) is what the code looked like previously. It handled both uninstall and install events with the same receiver using `when`. One of your commits seems to have separated them into two different broadcast receivers. Is there a reason for it?

Reply:
I will use a single receiver in this case. There is no need for two.